app/code/core/Mage/Checkout/etc/system.xml

```diff
@@ -45,16 +45,15 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                         </guest_checkout>
-                        <customer_must_be_logged translate="label">
-                            <label>Require Customer To Be Logged In To Checkout</label>
+                        <redirect_register translate="label">
```

Comment:
It's BC too if we keep the same name:

1) Store owner at some point had disabled guest checkout, and enabled `customer_must_be_logged`.
    * Customers will be redirected to register page.
2) Store owner re-enables guest checkout.
    * The `customer_must_be_logged` config option is hidden, but the value remains in the DB as true.
    * Customers *will not be* redirected to register page, because the setting only has effect when guest checkout is disabled.
3) Store owner upgrades to Maho
    * We now read that field without checking if guest checkout is enabled or not.
    * Customers will be redirected to register page.

I think the benefits of renaming outweigh keeping the old name. Because `customer_must_be_logged` doesn't make any sense if they don't need to be logged in if using guest checkout.

Reply:
The non BC solution I can think of is to use an upgrade script setting `redirect_register`'s value to true if both guest checkout is disabled, and `customer_must_be_logged` is enabled, for each website. Do you think that's worth it?